SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialReceiver.cpp

```diff
@@ -350,8 +369,19 @@ void USpatialReceiver::OnAuthorityChange(const Worker_AuthorityChangeOp& Op)
 		return;
 	}
 
+	// Update this worker's view of authority. We do this here as this is when the worker is first notified of the authority change.
+	// This way systems that depend on having non-stale state can function correctly.
+	StaticComponentView->OnAuthorityChange(Op);
```

Comment:
This should happen before the async stuff above.

Reply:
And in `HandleQueuedOpForAsyncLoad`, we should just do `HandleActorAuthority` instead of `OnAuthorityChange` when processing the queued ops.